src/vcpkg/paragraphs.cpp

```diff
@@ -420,55 +419,48 @@ namespace vcpkg::Paragraphs
 
         const auto manifest_path = port_directory / "vcpkg.json";
         const auto control_path = port_directory / "CONTROL";
-        const auto port_name = port_directory.filename().to_string();
+        const auto port_name = port_directory.filename();
         std::error_code ec;
         auto manifest_contents = fs.read_contents(manifest_path, ec);
         if (ec)
         {
-            if (fs.exists(manifest_path, IgnoreErrors{}))
+            const auto exists = ec != std::errc::no_such_file_or_directory;
+            if (exists)
             {
-                auto error_info = std::make_unique<ParseControlErrorInfo>();
-                error_info->name = port_name;
-                error_info->error = msg::format_error(msgFailedToParseManifest, msg::path = manifest_path);
-                return error_info;
+                return ParseControlErrorInfo::from_error(
+                    port_name, msg::format_error(msgFailedToParseManifest, msg::path = manifest_path));
```

Comment:
Noting that this should probably return the underlying filesystem error (permission denied) instead of a generic "failed to parse".

No change requested in this PR.

Reply:
The other place that used this message indeed included the original message, so I added it here.